pyfakefs/fake_filesystem.py

```diff
@@ -212,6 +218,7 @@ def __init__(
         self.path_separator: str = path_separator
         self.alternative_path_separator: Optional[str] = os.path.altsep
         self.patcher = patcher
+        self.create_temp_dir = create_temp_dir
```

Comment:
This is called twice in__init__, here and at line 257. Why is does the temp dir need to be created twice?

Reply:
Actually no - I'm just setting the attribute here, while on ine 257 I call the method (maybe the naming could be better). The attribute is needed to know if the temp dir shall be recreated in `reset`, and I check for the attribute in the method `_create_temp_dir` to make sure it is only recreated if it had been created before.